charts/sn-platform/templates/_helpers.tpl

```diff
@@ -157,4 +157,56 @@ jvmOptions:
   gcLoggingOptions:
   {{- toYaml . | nindent 2 }}
   {{- end }}
+{{- end }}
+
+{{/*
+Define function for save authenticaiton provider list
+*/}}
+{{- define "pulsar.authenticationProviders" }}
+{{- $authenticationProviders := list -}}
+{{- if .Values.auth.vault.enabled }}
+{{- $authenticationProviders = append $authenticationProviders "io.streamnative.pulsar.broker.authentication.AuthenticationProviderOIDCToken" }}
+{{- end }}
+{{- if .Values.auth.authentication.tls.enabled }}
+{{- $authenticationProviders = append $authenticationProviders "org.apache.pulsar.broker.authentication.AuthenticationProviderTls" }}
+{{- end }}
+{{- if .Values.auth.oauth.enabled }}
+{{- $authenticationProviders = append $authenticationProviders "io.streamnative.pulsar.broker.authentication.AuthenticationProviderOAuth" }}
+{{- end }}
+{{- join "," (compact $authenticationProviders) | quote }}
+{{- end }}
+
+{{/*
+Define function for save authenticaiton configuration
+*/}}
+{{- define "pulsar.autoConfiguration" }}
+{{- if .Values.auth.vault.enabled }}
+brokerClientAuthenticationPlugin: "org.apache.pulsar.client.impl.auth.AuthenticationToken"
```

Comment:
What if multiple auths are enabled?

Reply:
That will use the last brokerClientAuthenticationPlugin configuration